### DISCUSSION TITLE
How can I execute an SQL script using an async engine?

### DISCUSSION THREAD

**Discussion by ghevge** [sqlalchemy/sqlalchemy#6653] [Usage Questions]
For a sync engine , I was able to run any SQL query calling the .execute method. With an async  engine, that doesn't work anymore. Is it still possible to run SQL queries using async engines? If yes where can I find a sample of this ? in the docs, I was not able to locate an example https://docs.sqlalchemy.org/en/14/orm/extensions/asyncio.html:

`async with db_engine.connect() as db_conn:
        return (await db_conn.execute("""
            SELECT NOT EXISTS (
                SELECT * FROM information_schema.tables
                WHERE table_name = 'alembic_version'
            );
        """)).scalar()`

for which I get the error: 

`File "/bolt-webserver/scripts/initdb.py", line 124, in is_db_empty
    return (await db_conn.execute("""
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/ext/asyncio/engine.py", line 405, in execute
    result = await greenlet_spawn(
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/util/_concurrency_py3k.py", line 114, in greenlet_spawn
    result = context.switch(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1578, in _execute_20
    util.raise_(
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception
sqlalchemy.exc.ObjectNotExecutableError: Not an executable object: "\n            SELECT NOT EXISTS (\n                SELECT * FROM information_schema.tables\n                WHERE table_name = 'alembic_version'\n            );\n        "
ERROR: 1`

**Comment by CaselIT** [comment]
Hi,

The sync engine in 1.4 will warn telling you to use `execute(text(sql))` or `exec_driver_sql(sql)`.

The async engine has only the new interface, so no warning is issued, but it errors instead.

A better error is probably appropriate
